src/txn/txn_rollback_to_stable.c

```diff
@@ -1138,6 +1138,30 @@ __rollback_to_stable_hs_final_pass(WT_SESSION_IMPL *session, wt_timestamp_t roll
     return (ret);
 }
 
+/*
```

Comment:
I feel this function should be in txn_recover.c file and setting of the `last_ckpt_base_write_gen` must be done whenever the recovery function gets entered after the following code. In later tickets, we will move the RTS to this location to let the cleanup happens before the base write generation number is incremented.

>      F_SET(conn, WT_CONN_RECOVERING);

Reply:
Yeah, I remember we discussed moving the RTS before calling recovery. Updated the code to set `last_ckpt_base_write_gen` before the base write generation number is incremented.